src/service/resource.js

```diff
@@ -103,6 +104,15 @@ export class Resource {
       Resource.forElementOptional(element).updateOwner(owner);
     }
     element[OWNER_PROP_] = owner;
+    // Need to clear owner cache for all child elements
```

Comment:
We definitely want this to be recursive, as it could be `<owner> <child> <div> <grandchild>`. Using `Resources#discoverResourcesForElement_` or `#discoverResourcesForArray_` would do it.